openwisp_users/base/models.py

```diff
@@ -129,6 +129,25 @@ def clean(self):
         if self.phone_number == '':
             self.phone_number = None
 
+    @property
+    def permissions(self):
+        cache_key = f'user_{self.pk}_permissions'
+        permissions = cache.get(cache_key)
+        if permissions is not None:
+            return permissions
+        permissions = self.get_all_permissions()
+        cache.set(cache_key, permissions)
+        return permissions
+
+    def has_model_perm(self, app_label, model):
+        for perm in self.permissions:
+            if (
+                app_label.lower() == perm.split('.')[0]
+                and model.lower() == perm.split('_')[-1]
+            ):
+                return True
+        return False
```

Comment:
Why don't we just do:

```
def has_permission(self, permission):
    return permission in self.permissions
```

Since in openwisp-utils, we get the permission like `{app_label}.{model}`.

This method needs a brief mention in the README after `User permissions`